File: C++/BinarySearch/sqrt.cpp
#include<iostream>
using namespace std;
#define EPS 0.00001
double sqrt(double n) {
	double lo = 0, hi = n/2+1;
	double mid, sq;
	while(hi-lo > EPS) {
		mid = lo + (hi-lo)/2;
		sq = mid * mid;
		if(sq < n)
			lo = mid;
		else
			hi = mid;
	}
	return lo + (hi-lo)/2;
}

int main() {
	double n;
	cin >> n;
	cout << sqrt(n) << "\n";
	return 0;
}

File: Java/src/BinarySearch/Sqrt.java
package BinarySearch;

import FastScanner.FastScanner;

import java.io.BufferedOutputStream;
import java.io.PrintWriter;

public class Sqrt {
    static FastScanner fs;
    static PrintWriter out;
    public static double EPS = 0.0001;
   
    private static double sqrt(double n) {
        double lo = 0, hi = n/2 + 1, mid = lo;
        while(hi - lo > EPS) {
            mid = lo + (hi-lo)/2;
            double sq = mid * mid;
            if(sq > n) {
                hi = mid;
            } else {
                lo = mid;
            }
        }
        return mid;
    }
    
    public static void main(String[] args) {
        fs = new FastScanner();
        out = new PrintWriter(new BufferedOutputStream(System.out));
        double n = fs.nextDouble();
        out.println(sqrt(n));;
        out.close();
    }
}
